### PR TITLE
Feature: Adds a number of queries, mutations & schema updates for use with the `CLI`

### PR DESCRIPTION
**Pull request by CGoodwin90**
Adds a number of queries, mutations & schema updates required for https://github.com/uselagoon/lagoon-cli/pull/321

Companion PR to https://github.com/uselagoon/lagoon-cli/pull/321

### CODE REVIEW

**Comment by shreddedbacon** [comment]
Is this still WIP?

File: api/lagoon/client/_lgraphql/tasks/GetInvokableAdvancedTaskDefinitionsByEnvironment.graphql
Comment (shreddedbacon) [inline api/lagoon/client/_lgraphql/tasks/GetInvokableAdvancedTaskDefinitionsByEnvironment.graphql:1]: rename to `get..`

File: api/lagoon/client/projects.go [hunk 1]
```diff
@@ -242,3 +242,50 @@ func (c *Client) RemoveProjectFromOrganization(ctx context.Context, in *schema.R
 		Response: out,
 	})
 }
+
+// ProjectKeyByName queries the Lagoon API for a project by its name, and returns the public key & optionally the private key.
+func (c *Client) ProjectKeyByName(ctx context.Context, name string, revealValue bool, project *schema.Project) error {
```
Comment (shreddedbacon) [inline api/lagoon/client/projects.go:247]: `revealValue` -> `revealKey`

File: api/lagoon/client/tasks.go [hunk 2]
```diff
@@ -148,3 +148,95 @@ func (c *Client) UploadFilesForTask(ctx context.Context,
...
+	return c.client.Run(ctx, req, &struct {
+		Response *schema.Environment `json:"environmentByName"`
+	}{
+		Response: environment,
+	})
+}
+
+// InvokableAdvancedTaskDefinitionsByEnvironment gets tasks for an environment.
+func (c *Client) InvokableAdvancedTaskDefinitionsByEnvironment(ctx context.Context, projectID uint, environmentName string, environment *schema.Environment) error {
+	req, err := c.newRequest("_lgraphql/tasks/GetInvokableAdvancedTaskDefinitionsByEnvironment.graphql",
```
Comment (shreddedbacon) [inline api/lagoon/client/tasks.go:172]: After renaming the file, `getInvokable...`

File: api/lagoon/client/usergroups.go [hunk 3]
```diff
@@ -400,3 +399,103 @@ func (c *Client) UsersByOrganizationName(ctx context.Context, name string, users
 	json.Unmarshal(data, users)
 	return nil
 }
+
+// AddProjectToGroup adds a user to a group.
```
Comment (shreddedbacon) [inline api/lagoon/client/usergroups.go:403]: adds a group to a project :see_no_evil:

File: api/lagoon/projects.go [hunk 4]
```diff
@@ -95,3 +98,21 @@ func RemoveProjectFromOrganization(ctx context.Context, in *schema.RemoveProject
 	response := schema.Project{}
 	return &response, out.RemoveProjectFromOrganization(ctx, in, &response)
 }
+
+// GetProjectKeyByName gets the project keys of a project in Lagoon.
+func GetProjectKeyByName(ctx context.Context, name string, revealValue bool, p Projects) (*schema.Project, error) {
```
Comment (shreddedbacon) [inline api/lagoon/projects.go:103]: `revealValue` -> `revealKey`

File: api/schema/project.go [hunk 5]
```diff
@@ -127,11 +126,17 @@ type UpdateProjectPatchInput struct {
 	DevelopmentBuildPriority     *uint                `json:"developmentBuildPriority,omitempty"`
 	DeploymentsDisabled          *uint                `json:"deploymentsDisabled,omitempty"`
 	// `null` is valid graphql, use a pointer to allow `nil` to be empty
-	BuildImage *null.String `json:"buildImage,omitempty"`
+	BuildImage              *string `json:"buildImage,omitempty"`
```
Comment (shreddedbacon) [inline api/schema/project.go:129]: There is a reason this was `*null.String` as it allows for a JSON value `null` to be provided in the mutation

`{ buildImage: null }` is different to `{ buildImage: "null" }` and the `null.String` allows you to use the first one to effectively "unset" a build image.

**Review by shreddedbacon — CHANGES_REQUESTED** [review]

File: api/schema/project.go (hunk 5 above)
Comment (CGoodwin90) [reply api/schema/project.go]: All updated now

File: api/schema/project.go [hunk 6]
```diff
@@ -1,7 +1,5 @@
 package schema
 
-import "github.com/guregu/null"
```
Comment (shreddedbacon) [inline api/schema/project.go:3]: Still need the import

**Review by shreddedbacon — APPROVED** [review]

**Pull request merged**